### ISSUE TITLE
[NFR] Switch block for Volt

### ISSUE THREAD

**Issue by lajosbencz** [phalcon/cphalcon#13107]
New feature request, as per: https://forum.phalconphp.com/discussion/17035/feature-request-switch-for-volt

# Intended behaviour

Allows for switch/case conditional rendering in Volt

# Example

```volt
{% switch matrixBlock.type %}

 {% case "text" %}

     {{ matrixBlock.textField | markdown }}

 {% case "image" %}

    {{ matrixBlock.image[0].getImg() }}

 {% default %}

    <p>A font walks into a bar.</p>
    <p>The bartender says, “Hey, we don’t serve your type in here!”</p>

{% endswitch %}
```

**Comment by JABirchall** [comment]
Good suggestion Would be great for example a role badge would need a way to break out the switch as well so that it doesn't cascade. I suggest a `{% casebreak 'match' %}`

```php
{% switch user.getRoles() %}
    {% case "Owner" %}
    <span class="label label-pill label-primary">Owner</span>
    {% case "Admin" %}
     <span class="label label-pill label-primary">Admin</span>
    {% casebreak "Partner" %} // this should break the switch
    <span class="label label-pill label-primary">Partner</span>
    {{ matrixBlock.image[0].getImg() }}
    {% default %}
    Member
{% endswitch %}
```

**Comment by Slyvampy** [comment]
Thanks for creating this NFR for me :)

**Comment by sergeyklay** [comment]
Actually I prefer more something like: 

```jinja
{% switch username %}
    {# Each `case` statement can take multiple values to compare the variable #}
    {% case "Jim" "Bob" "Joe" %}
        Hello {{ username }}
    {% case "Nik" %}
        Hey hey {{ username }}!
    {% else %}
        {#
            An optional `else` statement sets off the default output
            if no matches could be found
        #}
        Who are you?
{% endswitch %}
```

I'll try to deal with this, but note it will be a switch without `break`-statements implementation.

**Comment by Jurigag** [comment]
Imho there shouldn't be endcase. Break should be just detected by next `{% case` or `{% default`.

Also not sure about this multiple values for case. Is it possible in plain php? I don't think so, not sure if i saw syntax like this in any language/template. But it might be good thing.

**Comment by oscarmolinadev** [comment]
@Jurigag You "can" more or less with true/false.

```
<?php
        $value = 'Nik';
        switch (true) {
            case ($value === 'Jim' || $value === 'Bob' || $value === 'Joe'): {
                echo 'Hello Multiple';
            }
            break;
            case ($value === 'Nik') : {
                echo 'Hello Nik';
            }
            break;
            default: {
                echo 'Hello nothing';
            }
        }
```

**Comment by JABirchall** [comment]
@oscarmolinadev but that is an antipattern and not how a switch case is designed to be used. you might as well generate a series of if statements. the switch() is omitted how you are using it.

**Comment by oscarmolinadev** [comment]
@JABirchall Totally agree with you! My example is what you never should do. I see this ugly sh*** in my work one time. That's why I've put "can" :) In PHP you can make ugly things like that.. and `while(true){}` makes no sense  in PHP either.

**Comment by sergeyklay** [comment]
I'm almost done worked version for php5 (without break still) so I need few days for php7

**Comment by JABirchall** [comment]
>I see this ugly sh*** in my work one time. 

@oscarmolinadev Yea, if i see code like that in a PR at work I rip them a new one. It's not our policy to be rude to other employees when they do dumb shit like that but I just can't help my self xD. its something that infuriates me.

**Comment by Jurigag** [comment]
@oscarmolinadev this is not purpose of switch really. For that you have `if construction`

**Comment by sergeyklay** [comment]
Latest news: i have worked version for switch-case-break-default for php 5

**Comment by sergeyklay** [comment]
Implemented in 3.3.x branch
